### PR TITLE
[v1.x] ONNX fix softmax

### PR DESCRIPTION
**Pull request by waytrue17**
## Description ##
Fix onnx softmax to support `use_length` for Bert use case.
Add test case for softmax.
Update test pipeline to support multiple input data types. 

### Essentials ###
- [x] PR's title starts with a category (e.g. [BUGFIX], [MODEL], [TUTORIAL], [FEATURE], [DOC], etc)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage
- [ ] Code is well-documented

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

File: tests/python-pytest/onnx/test_operators.py [hunk 1]
```diff
@@ -205,3 +206,15 @@ def test_onnx_export_fully_connected(tmp_path, dtype, num_hidden, no_bias, flatt
...
         args.append(mx.nd.random.uniform(0,1,(num_hidden,)))
     op_export_test('FullyConnected', M, args, tmp_path)
+
+@pytest.mark.parametrize('dtype', ['float32', 'float64'])
+def test_onnx_export_softmax(tmp_path, dtype):
+    x = mx.nd.random.uniform(0, 1, (2, 3, 4), dtype=dtype)
+    M1 = def_model('softmax')
+    op_export_test('softmax_1', M1, [x], tmp_path)
+    M2 = def_model('softmax', use_length=True, axis=0)
+    l2 = mx.nd.array([[2,0,3,1],[1,3,2,0], [0,0,0,1]], dtype=int)
```
Comment (Zha0q1) [inline tests/python-pytest/onnx/test_operators.py:216]: Shall we also use randomized inputs here?

Comment (waytrue17) [reply tests/python-pytest/onnx/test_operators.py:216]: `l2` and `l3` are intentionally generated to test certain cases e.g. [0,0,0,1] and [0,0,0]. I think it should be fine as `x` is randomly generated.

Comment (Zha0q1) [reply tests/python-pytest/onnx/test_operators.py:216]: gotcha

File: tests/python-pytest/onnx/test_operators.py [hunk 2]
```diff
@@ -253,3 +254,16 @@ def test_onnx_export_dropout(tmp_path, dtype, p):
     M = def_model('Dropout', p=p)
     x = mx.nd.array([[3,0.5,-0.5,2,7],[2,-0.4,7,3,0.2]], dtype=dtype)
     op_export_test('Dropout', M, [x], tmp_path)
+
+@pytest.mark.parametrize('dtype', ['float16', 'float32'])
+def test_onnx_export_softmax(tmp_path, dtype):
+    x = mx.nd.random.uniform(0, 1, (2, 3, 4), dtype=dtype)
```
Comment (Zha0q1) [inline tests/python-pytest/onnx/test_operators.py:260]: sorry about nitpicking: shall we also test high dimensional + axis = middle dim cases?

Comment (Zha0q1) [reply tests/python-pytest/onnx/test_operators.py:260]: We might also parametrize temperature?

Comment (waytrue17) [reply tests/python-pytest/onnx/test_operators.py:261]: added more tests for temperature and axis

**Review by Zha0q1 — APPROVED** [review]
LGTM!

**Comment by waytrue17** [comment]
@mxnet-bot run ci [unix-gpu, unix-gpu-cu110]

**Review by josephevans — APPROVED** [review]

**Comment by Zha0q1** [comment]
Thanks!

**Pull request merged**